crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -372,12 +390,16 @@ impl WidgetMachine {
         request_id: Uuid,
         response: Result<MatrixDriverResponse, String>,
     ) -> Vec<Action> {
-        let Some(request) = self.pending_matrix_driver_requests.remove(&request_id) else {
-            error!("Received response for an unknown request");
-            return Vec::new();
-        };
-
-        request.response_fn.map(|response_fn| response_fn(response, self)).unwrap_or_default()
+        self.pending_matrix_driver_requests
+            .extract(&request_id)
+            .map_err(|e| warn!("{e}"))
+            .map(|request| {
+                request
+                    .response_fn
+                    .map(|response_fn| response_fn(response, self))
+                    .unwrap_or_default()
+            })
+            .unwrap_or_default()
```

Comment:
This combination of `map_err` + `map` + `unwrap_or_default` would probably be more comprehensible when written as a single `match`.

Reply:
You're right, fixed.